pkg/apis/pingcap/v1alpha1/tidbclusterautoscaler_types.go

```diff
@@ -123,40 +102,9 @@ type TidbAutoScalerSpec struct {
 // +k8s:openapi-gen=true
 // BasicAutoScalerSpec describes the basic spec for auto-scaling
 type BasicAutoScalerSpec struct {
-	// maxReplicas is the upper limit for the number of replicas to which the autoscaler can scale out.
-	// It cannot be less than minReplicas.
-	// Deprecated
-	// +optional
-	MaxReplicas *int32 `json:"maxReplicas,omitempty"`
-
-	// minReplicas is the lower limit for the number of replicas to which the autoscaler
-	// can scale down.  It defaults to 1 pod. Scaling is active as long as at least one metric value is
-	// available.
-	// Deprecated
-	// +optional
-	MinReplicas *int32 `json:"minReplicas,omitempty"`
-
 	// Rules defines the rules for auto-scaling with PD API
 	Rules map[corev1.ResourceName]AutoRule `json:"rules,omitempty"`
 
-	// ScaleInIntervalSeconds represents the duration seconds between each auto-scaling-in
-	// If not set, the default ScaleInIntervalSeconds will be set to 500
-	// +optional
-	ScaleInIntervalSeconds *int32 `json:"scaleInIntervalSeconds,omitempty"`
-
-	// ScaleOutIntervalSeconds represents the duration seconds between each auto-scaling-out
-	// If not set, the default ScaleOutIntervalSeconds will be set to 300
-	// +optional
-	ScaleOutIntervalSeconds *int32 `json:"scaleOutIntervalSeconds,omitempty"`
```

Comment:
These two fields should be preserved and the scaling should check against them.

Reply:
If some groups need scale in while some groups need scale out, is this considered scale in or scale out and which interval should we check against?

Reply:
The interval check should be per group, so we have to record the lastScalingTime of each group and then check the interval accordingly.